Source/DataSources/VelocityOrientationProperty.js

```diff
@@ -0,0 +1,217 @@
+
+                    if (defined(value)) {
+                        this._subscription = value._definitionChanged.addEventListener(function() {
+                            this._definitionChanged.raiseEvent(this);
+                        }, this);
+                    }
+
+                    this._definitionChanged.raiseEvent(this);
+                }
+            }
+        },
+        /**
+         * Gets or sets the ellipsoid used to determine which way is up.
+         * @memberof VelocityOrientationProperty.prototype
+         *
+         * @type {Property}
+         */
+        ellipsoid : {
+            get : function() {
+                return this._ellipsoid;
+            },
+            set : function(value) {
+                var oldValue = this._ellipsoid;
+                if (oldValue !== value) {
+                    this._ellipsoid = value;
+                    this._definitionChanged.raiseEvent(this);
+                }
+            }
+        }
+    });
+
+    var position1Scratch = new Cartesian3();
+    var position2Scratch = new Cartesian3();
+    var normalScratch = new Cartesian3();
+    var directionScratch = new Cartesian3();
+    var rightScratch = new Cartesian3();
+    var upScratch = new Cartesian3();
+    var timeScratch = new JulianDate();
+    var basisScratch = new Matrix3();
+    var step = 1.0 / 60.0;
+
+    /**
+     * Gets the value of the property at the provided time.
+     *
+     * @param {JulianDate} [time] The time for which to retrieve the value.
+     * @param {Quaternion} [result] The object to store the value into, if omitted, a new instance is created and returned.
+     * @returns {Quaternion} The modified result parameter or a new instance if the result parameter was not supplied.
+     */
+    VelocityOrientationProperty.prototype.getValue = function(time, result) {
+        //>>includeStart('debug', pragmas.debug);
+        if (!defined(time)) {
+            throw new DeveloperError('time is required');
+        }
+        //>>includeEnd('debug');
+
+        var property = this._position;
+        if (Property.isConstant(property)) {
+            return undefined;
+        }
+
+        var position1 = property.getValue(time, position1Scratch);
+        var position2 = property.getValue(JulianDate.addSeconds(time, step, timeScratch), position2Scratch);
+
+        //If we don't have a position for now, return undefined.
+        if (!defined(position1)) {
+            return undefined;
+        }
+
+        //If we don't have a position for now + step, see if we have a position for now - step.
+        if (!defined(position2)) {
+            position2 = position1;
+            position1 = property.getValue(JulianDate.addSeconds(time, -step, timeScratch), position2Scratch);
+
+            if (!defined(position1)) {
+                return undefined;
+            }
+        }
+
+        var normal = defaultValue(this._ellipsoid, Ellipsoid.WGS84).geodeticSurfaceNormal(position1, normalScratch);
```

Comment:
Can we move this code to a generic function (could even be `@private` for now) that given the ellipsoid, position, and vector, creates the 3x3 matrix?  Perhaps it belongs in [Transforms](https://github.com/AnalyticalGraphicsInc/cesium/blob/master/Source/Core/Transforms.js).  This is a farily common operation.

Also, are we sure this robustly handles all possible velocities, e.g., when the object is traveling along the surface normal?  I used the same algorithm for Santa one year and needed a special case for `Cartesian3.cross(direction, normal, rightScratch)`.